contracts/standard/arbitration/composed-arbitrable/example/ArbitrableTokenList.sol

```diff
@@ -0,0 +1,594 @@
+    }
+
+    /** @dev Executes the ruling on the specified agreement.
+     *  @param _agreementID The ID of the agreement.
+     *  @param _ruling The ruling.
+     */
+    function executeAgreementRuling(bytes32 _agreementID, uint _ruling) internal {
+        super.executeAgreementRuling(_agreementID, _ruling);
+        uint256 disputeID = agreements[_agreementID].disputeID;
+        Item storage item = items[disputeIDToItem[disputeID]];
+        Agreement storage agreement = agreements[_agreementID];
+        require(agreement.disputed, "The item is not disputed.");
+
+        if (_ruling == REGISTER) {
+            if (item.status == ItemStatus.Resubmitted || item.status == ItemStatus.Submitted)
+                agreement.parties[0].send(item.balance); // Deliberate use of send in order to not block the contract in case of reverting fallback.
+            else
+                agreement.parties[1].send(item.balance);
+
+            item.status = ItemStatus.Registered;
+        } else if (_ruling == CLEAR) {
+            if (item.status == ItemStatus.PreventiveClearingRequested || item.status == ItemStatus.ClearingRequested)
+                agreement.parties[0].send(item.balance);
+            else
+                agreement.parties[1].send(item.balance);
+
+            item.status = ItemStatus.Cleared;
+        } else { // Split the balance 50-50 and give the item the initial status.
+            if (item.status == ItemStatus.Resubmitted)
+                item.status = ItemStatus.Cleared;
+            else if (item.status == ItemStatus.ClearingRequested)
+                item.status = ItemStatus.Registered;
+            else
+                item.status = ItemStatus.Absent;
+
+            agreement.parties[0].send(item.balance / 2);
+            agreement.parties[1].send(item.balance / 2);
+        }
+
+        agreement.disputed = false;
+        item.balance = 0;
+        agreement.parties[1] = 0x0; // Dispute has been resolved, reset challenger.
+        item.challengeReward = 0; // Clear challengeReward once a dispute is resolved.
+
+        emit ItemStatusChange(agreement.parties[0], agreement.parties[1], disputeIDToItem[disputeID], item.status, agreement.disputed);
+    }
+
+    /* Interface Views */
+
+    /**
+     *  @dev Return the number of items in the list.
+     *  @return The number of items in the list.
+     */
+    function itemsCount() public view returns (uint count) {
+        count = itemsList.length;
+    }
+
+    /**
+     *  @dev Return the numbers of items in the list per status.
+     *  @return The numbers of items in the list per status.
+     */
+    function itemsCounts() public view returns (uint pending, uint challenged, uint accepted, uint rejected) {
+        for (uint i = 0; i < itemsList.length; i++) {
+            Item storage item = items[itemsList[i]];
+            Agreement storage latestAgreement = agreements[latestAgreementId(itemsList[i])];
+            if (latestAgreement.disputed) challenged++;
```

Comment:
Let's map these counts to the actual statuses in the enum. So the front end can decide what is what. Some use cases might require differentiating between Resubmitted and Submitted, etc.

Reply:
Can be external.